### ISSUE TITLE
Recursive exceptions on manager shutdown

### ISSUE THREAD

**Issue by shuds13** [Libensemble/libensemble#396]
When the manager receives a worker exception, it will post stop signals to all workers and then process the `finally` block in libE_manager.py.  This calls `_final_receive_and_kill` which continues to loop round active workers receiving messages until they are closed out (unless wall-clock is exceeded).

If multiple workers have sent exceptions, then this will create recursive exception handling. I plan
to implement either:

* Only the first worker exception is raised - each worker is still checked and a received exception will still close it out.

or:

* The manager closes without polling the other workers once an exception is received.

Either of these will prevent the recursive exceptions and an option can easily be added (e.g. poll_on_wrk_exception). We should decide on the default. 

Note that the second option could exit without receiving some valid data, while the first could hang waiting on a long running/hanging simulation or if some other unexpected behavior prevents the worker returning. Both will still exit libE clean, as each call to libE now uses, and closes, it's own duplicate of the provided communicator.
